### DISCUSSION TITLE
Can the search results webpart Open files from Search results in Desktop application as per document library setting

### DISCUSSION THREAD

**Discussion by samg365** [microsoft-search/pnp-modern-search#2662] [Q&A] [answered]
Can the search results webpart Open files from Search results in Desktop application as per document libraries?
I understand this will not be possible for PDFs and other file types 
But they seem to be opening only in web version when opening Word, Excel, etc...  Even when the document library has the Open in the client application set on.

Any guidance on this would be appreciated.
Thank you.

**Comment by wobba** [comment, marked as answer]
You can create a custom web component which will call into SharePoint and find the default value for a library and use that, but I expect it to not be very performant as you have to check for each site you have a result from. You can however change the template used and craft your own URL's to open directly in a client. There should be information on the internet how to do this. But it only works if the client has Office installed.

**Reply by samg365** [reply]
Thanks @wobba will do some investigation and see what we can find on the 2nd option. If we are successful I'll add an update.

**Reply by samg365** [reply]
Here's how we modified the crafting of the urls to force them to open in the desktop.

<a href="
{{#if (eq (slot item @root.slots.FileType) "docx") }}
ms-word:ofe|u|
{{else if (eq (slot item @root.slots.FileType) "xlsx") }}
ms-excel:ofe|u|
{{else if (eq (slot item @root.slots.FileType) "pptx") }}
ms-powerpoint:ofe|u|
{{else}}
{{/if}}
{{slot item @root.slots.PreviewUrl}}"  target="_blank" style="color: {{@root.theme.semanticColors.link}}">	
{{slot item @root.slots.Title}}
</a>

Thanks for pointing me in the right direction.

**Reply by eric-duncan-pdx** [reply]
@samg365, can you share the complete results template you're using?  This is awesome work and I've often thought about how to accomplish the same thing with other solutions.

**Reply by samg365** [reply]
@eric-duncan-pdx I am not sure how to export the complete results template. Not something I've done before.
